test/extended/workspacetype/workspace.go

```diff
@@ -16,49 +18,90 @@ var _ = g.Describe("[sig-workspace]", func() {
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(output).Should(o.ContainSubstring("deleted"))
+			o.Eventually(func() string {
+				workSpaces, _ := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+childWorkspace.ParentServerURL, "workspace").Output()
+				return workSpaces
+			}, 60*time.Second, 5*time.Second).ShouldNot(o.ContainSubstring(childWorkspace.Name))
+		}
+
+		g.By("# Delete parent workspace")
+		output, err := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("delete").Args("--server="+parentWorkSpace.ParentServerURL, "workspace", parentWorkSpace.Name).Output()
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(output).Should(o.ContainSubstring("deleted"))
 		o.Eventually(func() string {
-			workSpaces, _ := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+myWorkSpace.ServerURL, "workspace").Output()
+			workSpaces, _ := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+parentWorkSpace.ParentServerURL, "workspace").Output()
 			return workSpaces
-		}, 60*time.Second, 5*time.Second).ShouldNot(o.ContainSubstring(mySubWorkSpace.Name))
+		}, 60*time.Second, 5*time.Second).ShouldNot(o.ContainSubstring(parentWorkSpace.Name))
+	})
+
+	// author: zxiao@redhat.com
+	g.It("Author:zxiao-Medium-[Serial] I can create context for a specific workspace and use it", func() {
+		g.By("# Create a test workspace")
+		k.SetupWorkSpace()
+		workSpace := k.WorkSpace()
 
-		// Delete the test workspace under home workspace and check it deleted successfully
-		output, err = k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("delete").Args("--server="+myWorkSpace.ParentServerURL, "workspace", myWorkSpace.Name).Output()
+		g.By("# Get context before test run")
+		contextBeforeTestRun, err := k.WithoutNamespace().WithoutKubeconf().Run("config").Args("current-context").Output()
 		o.Expect(err).NotTo(o.HaveOccurred())
-		o.Expect(output).Should(o.ContainSubstring("deleted"))
-		o.Eventually(func() string {
-			workSpaces, _ := k.WithoutNamespace().WithoutKubeconf().WithoutWorkSpaceServer().Run("get").Args("--server="+myWorkSpace.ParentServerURL, "workspace").Output()
-			return workSpaces
-		}, 60*time.Second, 5*time.Second).ShouldNot(o.ContainSubstring(myWorkSpace.Name))
+
+		g.By("# Create context under this workspace")
+		err = k.WithoutNamespace().WithoutKubeconf().Run("kcp").Args("workspace", "create-context", workSpace.Name, "--server="+workSpace.ParentServerURL).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("# Switch to the newly created context")
+		// NOTE: not doing assertion as to avoid double assertion failures, use DeferCleanup in ginkgo v2.0 in the future
+		defer func() {
+			k.WithoutNamespace().WithoutKubeconf().Run("config").Args("use-context", contextBeforeTestRun).Execute()
```

Comment:
Assertion for `delete-context` can be removed, but assertion for `use-context` is required.